modules/javafx.graphics/src/main/java/javafx/scene/layout/BorderPane.java

```diff
@@ -400,14 +403,16 @@ public final ObjectProperty<Node> rightProperty() {
             centerMinWidth = getAreaWidth(getCenter(), -1, true);
         }
 
-        final Insets insets = getInsets();
         return insets.getLeft() +
                 Math.max(leftPrefWidth + centerMinWidth + rightPrefWidth, Math.max(topMinWidth,bottomMinWidth)) +
                 insets.getRight();
     }
 
     @Override protected double computeMinHeight(double width) {
         final Insets insets = getInsets();
+        if (width != -1) {
+            width -= (insets.getLeft() + insets.getRight());
```

Comment:
Let's say we call `computeMinHeight(10)`, but the left and right insets are 20. This means that `width` is now -10, which probably means "ignore the value" (the spec isn't entirely clear about that, but -10 is not a valid width in any case).

I think the following code might be better:
```java
if (width >= 0) {
    width = Math.max(0, width - insets.getLeft() - insets.getRight());
}
```

Reply:
The `width` value passed to `computeMinHeight` (if not -1) should be the result of a call to `computeMinWidth(-1)` (depending on the result of `getContentBias`; this is specified somewhere); if that function always includes the insets, then subtracting them here should not result in a negative value.

Of course, one can never be too careful.

I don't like the changing of the meaning of the `width` parameter here though using parameter assignment.